moshi/src/test/java/com/squareup/moshi/KotlinExtensionsTest.kt

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2018 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.squareup.moshi
+
+import org.assertj.core.api.Assertions.assertThatCode
+import org.junit.Test
+
+class KotlinExtensionsTest {
+
+  @Test fun reifiedAdapter() {
+    val moshi = Moshi.Builder().build()
+
+    assertThatCode { moshi.adapter<TestUtil.Empty>() }.doesNotThrowAnyException()
```

Comment:
nit: can we smoke test a simple happy path here? (as opposed to just making sure it doesn't crash, we want to also make sure it still works)